python_modules/dagster/dagster/_core/definitions/asset_key.py

```diff
@@ -39,13 +40,15 @@ class AssetKey(NamedTuple("_AssetKey", [("path", PublicAttr[Sequence[str]])])):
             strings represent the hierarchical structure of the asset_key.
     """
 
-    def __new__(cls, path: Union[str, Sequence[str]]):
+    path: PublicAttr[List[str]]
```

Comment:
keeping this `Sequence` is a significant hit
https://docs.pydantic.dev/latest/concepts/performance/#sequence-vs-list-or-tuple-mapping-vs-dict